test/ansible/run_test.yaml

```diff
@@ -20,254 +16,18 @@
-          echo "done"
-        done
-
-  - name: create bmhs
-    shell: chmod +x /tmp/create_bmhs && /tmp/create_bmhs
-  - name: wait until BMHs
-    shell: "until kubectl get -n {{ test_namespace }} bmh; do sleep 1; done"
-  - name: wait for BMHs to be available
-    shell: "kubectl -n {{ test_namespace }} wait --for=jsonpath='{.status.provisioning.state}'=available bmh/bmh-vm-{{ item }}"
-    with_sequence: count="{{ number_of_nodes }}" format="%02x"
-
-  # setup test env vars
-  - name: Clean current test vars
-    command: rm -rf ~/.test-config
-  - name: Define SSH_AUTHORIZED_KEY
-    lineinfile:
-      path: ~/.test-config
-      line: "export SSH_AUTHORIZED_KEY='{{ ssh_authorized_key }}'"
-      create: true
-      state: present
-  - name: Define PULLSECRET
-    lineinfile:
-      path: ~/.test-config
-      line: "export PULLSECRET='{{ pullsecret }}'"
-      create: true
-      state: present
-
-  # create copy example manifest
-  - name: generate manifest with vars
-    shell: source ~/.test-config && cat "{{ src_dir }}/examples/{{ example_manifest }}" | sed 's|<PULLSECRET>|'"${PULLSECRET}"'|g' | sed 's|<SSH_AUTHORIZED_KEY>|'"${SSH_AUTHORIZED_KEY}"'|g' > /tmp/example_manifest.yaml
-
-  - name: apply example
-    shell: kubectl -n {{ test_namespace }} apply -f /tmp/example_manifest.yaml
-  - name: wait for ACI
-    shell: "until kubectl get -n {{ test_namespace }} aci {{ cluster_name }}; do sleep 1; done"
-
-  - name: wait for ACI to be installing
-    shell: "kubectl -n {{ test_namespace }} wait --for=jsonpath='{.status.debugInfo.state}'=installing aci/{{ cluster_name }} --timeout=1200s"
-
-  - name: wait controlplane to be ready
-    shell: "kubectl wait -n {{ test_namespace }} openshiftassistedcontrolplane/{{ cluster_name }} --for condition=Ready --timeout 3600s"
-
-  - name: make sure ACI to be installed
-    shell: "kubectl -n {{ test_namespace }} wait --for=jsonpath='{.status.debugInfo.state}'=adding-hosts aci/{{ cluster_name }} --timeout=3600s"
-
-  - name: upgrade cluster
-    shell: "kubectl -n {{ test_namespace }} patch openshiftassistedcontrolplane {{ cluster_name }} --type=merge --patch '{\"spec\": {\"distributionVersion\": \"{{ upgrade_to_version }}\"}}'"
-
-  - name: wait for upgrade to kick-in
-    shell: "sleep 60"
-
-  # wait ctrlplane to complete upgrade
-  - name: wait controlplane to complete upgrade
-    shell: "kubectl wait -n {{ test_namespace }} openshiftassistedcontrolplane/{{ cluster_name }} --for condition=UpgradeCompleted --timeout 3600s"
-
-  # wait ctrlplane to be ready
-  - name: wait controlplane to be ready
-    shell: "kubectl wait -n {{ test_namespace }} openshiftassistedcontrolplane/{{ cluster_name }} --for condition=Ready --timeout 7200s"
+    upgrade_to_version: "4.19.0-ec.3"
```

Comment:
this depends on https://github.com/openshift-assisted/cluster-api-provider-openshift-assisted/pull/102 to be merged